lib/request.ex

```diff
@@ -0,0 +1,12 @@
+defmodule Instagram.Request do
+  use HTTPoison.Base
+
+  @url "https://api.instagram.com/v1"
+  def process_url(url) do
+    case String.downcase(url) do
+      <<"http://"::utf8, _::binary>> -> url
+      <<"https://"::utf8, _::binary>> -> url
```

Comment:
Backwards compatible here. `https://` start to the string returns the url. 

`Instagram.Request.get!("https://api.instagram.com/v1/users/self/media/recent")`